### PR TITLE
manifest: Update Matter SDK revision to pull fixes from v2.4.1

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
- Pull all fixes provided within NCS 2.4.1 release to the Main branch.
- Update the connectedhomeip revision to pull all changes provided to NCS 2.4.1 tag.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -107,7 +108,9 @@ The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:108]: You need to update the SHA in line 107 whenever Matter SDK is updated.
See for example: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/releases/release-notes-2.2.0.html#matter-fork

**Review by greg-fer — CHANGES_REQUESTED** [review]
Additionally, we need to update the links in links.txt under the header ``### Matter links that need updated SHA per release``, currently line 125.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (ArekBalysNordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:108]: Thanks for the reminder. I will update it after merging the PR in SDK.

Comment (ArekBalysNordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:108]: Wait, @greg-fer we need to update the SHA after the alignment of sdk_connectedhomeip to a specific version of the Matter Upstream SDK. In this PR the `v1.1.0.1` version of the Matter SDK is still correct.

Comment (greg-fer) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:108]: Good to know, and sorry for the hassle. :)

**Review by LuDuda — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**